MediaBrowser.Model/Dlna/StreamBuilder.cs

```diff
@@ -2257,5 +2181,117 @@ private int GetRank(ref TranscodeReason a, TranscodeReason[] rankings)
 
             return index;
         }
+
+        private TranscodeReason GetCompatibilityContainer(MediaOptions options, MediaSourceInfo mediaSource, string container, MediaStream? videoStream)
+        {
+            var profile = options.Profile;
+
+            // Video
+            int? width = videoStream?.Width;
+            int? height = videoStream?.Height;
+            int? bitDepth = videoStream?.BitDepth;
+            int? videoBitrate = videoStream?.BitRate;
+            double? videoLevel = videoStream?.Level;
+            string? videoProfile = videoStream?.Profile;
+            VideoRangeType? videoRangeType = videoStream?.VideoRangeType;
+            float videoFramerate = videoStream is null ? 0 : videoStream.ReferenceFrameRate ?? 0;
+            bool? isAnamorphic = videoStream?.IsAnamorphic;
+            bool? isInterlaced = videoStream?.IsInterlaced;
+            string? videoCodecTag = videoStream?.CodecTag;
+            bool? isAvc = videoStream?.IsAVC;
+
+            TransportStreamTimestamp? timestamp = videoStream is null ? TransportStreamTimestamp.None : mediaSource.Timestamp;
+            int? packetLength = videoStream?.PacketLength;
+            int? refFrames = videoStream?.RefFrames;
+
+            int? numAudioStreams = mediaSource.GetStreamCount(MediaStreamType.Audio);
+            int? numVideoStreams = mediaSource.GetStreamCount(MediaStreamType.Video);
+
+            var checkVideoConditions = (ProfileCondition[] conditions) =>
```

Comment:
```suggestion
            IEnumerable<ProfileCondition> CheckVideoConditions(ProfileCondition[] conditions) =>
```

Nitpicking: I prefer this way so that it is a clear local function

Reply:
> We can extract it as well - it is used in 3 places.
> The only problem is that it will [collect parameters](https://github.com/jellyfin/jellyfin/blob/45f971035f7d2b4a583cac87a63e6ae28e80e383/MediaBrowser.Model/Dlna/StreamBuilder.cs#L2190-L2208) on each call.

Like this: https://github.com/dmitrylyzo/jellyfin/commit/99559f1a2984d0ad0607d703b19b01874040cede